Automattic/jetpack
# Protect: record event just after the site is registered

This PR register the Plugin added event just right after the Site registers, to handle the case that happens when the user activates the plugin from the interstitial page:

<img width="600" alt="image" src="https://user-images.githubusercontent.com/77539/166678912-f8a4e32e-1565-43a2-8cfd-6859403b1c48.png">

On thet page above, the site is not registered so recording event is not possible to do. However, and taking advantage of the Promise that the `handleRegisterSite()` helper could return, we can record the event just right after the site registers, in code words:

```es6
//...
// Record event in case the site did register.
handleRegisterSite().then( () =>
	recordEvent( 'jetpack_protect_offer_connect_product_activated' )
);
//...
```

Fixes #

#### Changes proposed in this Pull Request:

* Protect: record even just after the site is registered

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



Yes. It records an event right after the Protect plugin is added to the site

#### Testing instructions:






### Testing useConnection() hook

This PR touches the useConnection() hook, and although it should not introduce breaking changes, it's good to take a look at other pkgs/plugins that use it, for instance, test:

### Testing recording events

* Go to Protect plugin page
* Disconnected the site, in case it's connected
* Set localstorage debug: `localStorage.setItem( 'debug', 'dops:analytics*' );`
* Add the plugin
* Confirm it records the event `jetpack_protect_offer_connect_product_activated `

https://user-images.githubusercontent.com/77539/172691077-c2ac05eb-be23-4a58-b996-410f81ee20e6.mov


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202217068684601
